numpy/lib/tests/test_nanfunctions.py

```diff
@@ -1265,6 +1265,14 @@ def test_regression(self, weighted):
                      np.nanpercentile(ar, q=[50], axis=1, **w_args))
         assert_equal(np.nanquantile(ar, q=[0.25, 0.5, 0.75], axis=1, **w_args),
                      np.nanpercentile(ar, q=[25, 50, 75], axis=1, **w_args))
+    
+    @pytest.mark.parametrize(argnames='axis', 
+        argvalues=[1, (1, ), 2, (0, 2), -1, (-1, -2)])
+    def test_consistency_with_quantile(self, axis):
+        a = np.arange(42).reshape(3, 7, 2).astype(float)
```

Comment:
There are different code path for `a.ndim==1` and `a.ndim > 1`, so could parametrize the shape to cover this?

Reply:
@lorentzenchr Thank you for your helpful comments. I reckon the following should give adequate coverage? What do you think ? Based on your suggestions I propose adding assert that checks for equality alongside shapes and have parametrized over shape as well.
```
@pytest.mark.parametrize(argnames='axis',
     argvalues=[None, 0, 1, (1, ), 2, (0, 2), -1, (-1, -2)])
@pytest.mark.parametrize(argnames='shape',
                         argvalues=[(42, ), (3, 7, 2)])
def test_consistency_with_quantile(self, shape, axis):
    a = np.arange(np.prod(shape)).reshape(shape).astype(float)
    if a.ndim == 1 and axis not in (None, 0, -1):
        pytest.skip(f"Skipping test for {a.ndim}-D array with incompatible axis {axis}")
    q = np.full((3, 2), 0.5)
    assert_equal(np.nanquantile(a, q, axis=axis).shape,
                  np.quantile(a, q, axis=axis).shape)
    assert_equal(np.nanquantile(a, q, axis=axis),
                  np.quantile(a, q, axis=axis))
```